### PR TITLE
refactor(services-auth-delegation-api): Update domains endpoint

### PR DESCRIPTION
**Pull request by saevarma**
[Notion task](https://www.notion.so/Update-domains-endpoint-to-return-all-domains-with-query-param-for-only-supporting-delegations-5b4798752dda45a682aeb4d8a4c98e1c?pvs=4)

## What

Update domains endpoint to return all domains and add query params to filter by delegation support.

## Why

To resolve domain logo in session history, we need to be able to query all domains not only those that are supporting delegations.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts [hunk 1]
```diff
@@ -44,16 +44,37 @@ export class DelegationResourcesService {
...
+
     const domains = await this.domainModel.findAll({
-      where: and(...this.apiScopeFilter({ user, prefix: 'scopes', direction })),
+      where: onlyDelegations
+        ? and(
+            ...this.apiScopeFilter({ user, prefix: 'scopes', direction }),
+            domainNameFilter,
+          )
+        : domainNameFilter,
       include: [
```
Comment (eirikurn) [inline libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts:73]: Should skip this whole join tree unless `onlyDelegations`?

**Review by eirikurn — APPROVED** [review]

File: libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts (hunk 1 above)
Comment (saevarma) [reply libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts:73]: It is already doing that, when `onlyDelegations=false` then it only adds the `name: domainNames` if provided.
```
onlyDelegations
        ? and(
            ...this.apiScopeFilter({ user, prefix: 'scopes', direction }),
            domainNameFilter,
          )
        : domainNameFilter,
```

File: apps/services/auth/delegation-api/src/app/domains/domains.controller.ts [hunk 2]
```diff
@@ -59,11 +66,36 @@ export class DomainsController {
...
+        domainNames: {
+          description: 'A list of domain names to filter by.',
+          required: false,
+          schema: {
+            type: 'array',
+            items: {
+              type: 'string',
+              example: ['@island.is', '@admin.island.is'],
+            },
+          },
```
[the remark this answers is not in the document]
Comment (saevarma) [reply apps/services/auth/delegation-api/src/app/domains/domains.controller.ts:89]: Yes it is. Rename query param to `domainName` but keeping the array in code as `domainNames`

File: libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts (hunk 1 above)
Comment (saevarma) [reply libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts]: You meant the `include`, yes considered skipped.

**Pull request merged**